### PR TITLE
Bottom Navbar UI

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## Description

- Introduces a Bottom navigation bar to the UI

## Reference

[VPN-2487: Create custom navigation tab bar component](https://mozilla-hub.atlassian.net/browse/VPN-2487)

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

File: src/ui/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml [hunk 1]
```diff
@@ -119,17 +119,13 @@ VPNFlickable {
 
                 onClicked: {
                     VPN.recordGleanEvent("deleteAccountRequested");
+                    navbar.visible = false
```
Comment (MattLichtenstein) [inline src/ui/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml:122]: Even though it is a somewhat common pattern in this project, I don't love referencing ancestor components in other files like this since it's very fragile, still considering other implementations

File: src/ui/settings/ViewGiveFeedback.qml [hunk 2]
```diff
@@ -375,7 +378,7 @@ Item {
                        anchors.top = undefined;
                        anchors.topMargin = undefined;
                        anchors.bottom= parent.bottom
-                       anchors.bottomMargin = VPNTheme.theme.windowMargin * 4
+                       anchors.bottomMargin = VPNTheme.theme.navBarHeightWithMargins
```
Comment (MattLichtenstein) [inline src/ui/settings/ViewGiveFeedback.qml:381]: For non-scrolling page views, we will have to remember to account for the navbar by anchoring the bottom margin of the layout to this value. Alternatively, we can create a new base component (maybe something like `SinglePageView`) that has this built in so we don't have to remember to add it for every new page view, and can be the root component of all of our pages. That's a decent amount of churn so I haven't done that yet, but would like to get some thoughts

File: translations/strings.yaml [hunk 3]
```diff
@@ -677,6 +677,17 @@ tutorialPopup:
   tutorialLeaveSubtitle: Looks like you haven’t finished the tutorial quite yet. Would you like to continue and finish now or quit and come back later?
   tutorialLeavePrimaryButtonLabel: Resume tutorial
 
+navBar:
+  homeTab:
+    value: Home
+    comment: Label for the tab on the nav bar that takes the user to the home screen
```
Comment (flodolo) [inline translations/strings.yaml:683]: übernit: `for the tab on the nav bar` is incredibly hard to parse for me (because of the amount of short words), can we go with `for the tab on the navigation bar`?

**Review by flodolo — APPROVED** [review]

File: nebula/ui/components/VPNBottomNavigationBarButton.qml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        radius: parent.height / 2
+        opacity: .2
+        color: VPNTheme.theme.white
+    }
+
+    Image {
+        id: image
+        anchors.centerIn: parent
+        sourceSize.height: 32
+        sourceSize.width: 32
```
Comment (flozia) [inline nebula/ui/components/VPNBottomNavigationBarButton.qml:36]: Nit: Maybe we could use `VPNTheme.theme.iconSize * 2` here.

File: src/ui/settings/ViewGiveFeedback.qml (hunk 2 above)
Comment (flozia) [reply src/ui/settings/ViewGiveFeedback.qml:381]: I like the idea and I’m also wondering if we rather should have all views in a `PageView` that can be potentially scrollable. If we feel like that is too much work for now that would be probably a good task to add to a design system epic.

File: nebula/ui/components/VPNBottomNavigationBar.qml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+        hoverEnabled: true
+    }
+
+    RowLayout {
+        id: layout
+        anchors.fill: parent
+        anchors.leftMargin: tabs.length === 3 ? 48 : 32
+        anchors.rightMargin: anchors.leftMargin
+
+        spacing: tabs.length === 3 ? 51.5 : 24
```
Comment (flozia) [inline nebula/ui/components/VPNBottomNavigationBar.qml:46]: We’ll probably won’t have less than three tabs, but in the spirit of making the layout dynamic: `tabs.length <= 3`? Also: Could we have variables for those to make it a bit easier to grasp how the layout adjust to a changing amount of tabs?

File: nebula/ui/components/VPNServerList.qml [hunk 6]
```diff
@@ -128,19 +125,31 @@ FocusScope {
...
             Repeater {
                 id: countriesRepeater
                 model: countriesModel
-                delegate: VPNServerCountry{}
+                delegate: VPNServerCountry{
+                    //Necessary because VPNClickableRow has hardcoded anchors (that we aren't using here because this is a layout)
+                    anchors.left: undefined
+                    anchors.right: undefined
+                    anchors.leftMargin: undefined
+                    anchors.rightMargin: undefined
```
Comment (flozia) [inline nebula/ui/components/VPNServerList.qml:144]: Nit: For setting multiple `anchors` properties I really like using:
```suggestion
                    anchors {
                        left: undefined
                        right: undefined
                        leftMargin: undefined
                        rightMargin: undefined
                    }
```

**Review by flozia — COMMENTED** [review]
The Navbar looks awesome. 🎉  Left some minor remarks and until it’s fully functional we should consider to hide it for now.

File: nebula/ui/components/VPNBottomNavigationBarButton.qml [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import QtQuick 2.0
```
Comment (MattLichtenstein) [inline nebula/ui/components/VPNBottomNavigationBarButton.qml:1]: needs source code header

File: nebula/ui/components/VPNBottomNavigationBar.qml [hunk 8]
```diff
@@ -0,0 +1,55 @@
+import QtQuick 2.0
```
Comment (MattLichtenstein) [inline nebula/ui/components/VPNBottomNavigationBar.qml:1]: needs source code header

File: nebula/ui/components/VPNBottomNavigationBar.qml [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        id: layout
+        anchors.fill: parent
+        anchors.leftMargin: tabs.length === 3 ? 48 : 32
+        anchors.rightMargin: anchors.leftMargin
+
+        spacing: tabs.length === 3 ? 51.5 : 24
+        data: tabs
+
+        Component.onCompleted: {
+            for(var i = 0 ; i < tabs.length; i++) {
```
Comment (lesleyjanenorton) [inline nebula/ui/components/VPNBottomNavigationBar.qml:50]: supernit: `let`

File: src/ui/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml (hunk 1 above)
Comment (lesleyjanenorton) [reply src/ui/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml:122]: You could also do this by adding an objectName to the parent Item {} in `ViewDeleteAccount.qml` and then adding something along the lines of `&& mainStackView.currentItem.objectName !== "viewDeleteAccountObjectName"` to toggle the nav bar visibility in main.qml.

File: src/ui/views/ViewContactUs.qml [hunk 10]
```diff
@@ -80,8 +80,7 @@ Item {
             id: vpnFlickable
             property var appRating
             property var feedbackCategory
-            flickContentHeight: col.childrenRect.height
-            interactive: flickContentHeight > height
+            flickContentHeight: col.implicitHeight + col.anchors.topMargin
```
Comment (lesleyjanenorton) [inline src/ui/views/ViewContactUs.qml:83]: binding loop

**Comment by MattLichtenstein** [comment]
> I like the idea and I’m also wondering if we rather should have all views in a PageView that can be potentially scrollable. If we feel like that is too much work for now that would be probably a good task to add to a design system epic.

@flozia That sounds like we should just make every page based on a VPNFlickable, regardless of whether or not we expect it to scroll. I am fine with this, maybe we just start using flickables for all pages going forward?

File: nebula/ui/components/VPNServerList.qml (hunk 6 above)
Comment (MattLichtenstein) [reply nebula/ui/components/VPNServerList.qml:144]: I have no preference either way - but I know this is whats recommended by Qt's own conventions. My vote is for consistency. If we can get buy-in from all UI engineering to do this going forward, I am for it. 

Maybe we should start a document on our own qml coding conventions?

File: src/ui/main.qml [hunk 11]
```diff
@@ -121,6 +121,77 @@ Window {
...
+        Behavior on opacity {
+            PropertyAnimation {
+                duration: 500
+            }
+        }
+
+        Connections {
+            target: VPNConnectionBenchmark
+            onStateChanged: {
+                navbar.opacity = VPNConnectionBenchmark.state === VPNConnectionBenchmark.StateInitial ? 1 : 0
```
[the remark this answers is not in the document]
Comment (MattLichtenstein) [reply src/ui/main.qml:147]: its disabled on zero opacity, so not accessible to screen readers 😄

File: src/ui/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml (hunk 1 above)
Comment (MattLichtenstein) [reply src/ui/settings/ViewSubscriptionManagement/ViewSubscriptionManagement.qml:122]: haha seems hacky, but probably better than all the rebinding

File: src/ui/main.qml [hunk 12]
```diff
@@ -121,6 +121,77 @@ Window {
...
+            }
+        }
+
+        Connections {
+            target: VPNAuthInApp
+            onStateChanged: {
+                if(VPNAuthInApp.state === VPNAuthInApp.StateInitializing) {
+                    navbar.visible = Qt.binding(() => VPN.state === VPN.StateMain)
```
[the remark this answers is not in the document]
Comment (MattLichtenstein) [reply src/ui/main.qml:155]: the binding may be broken on line 122 in ViewSubscriptionManagement.qml, but will make the change you suggested [here](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/4072#discussion_r933641617) and get rid of this
